node/service/src/service.rs

```diff
@@ -522,14 +525,16 @@ fn frequency_dev_instant(config: Configuration) -> Result<TaskManager, sc_servic
 		let (command_sink, commands_stream) = futures::channel::mpsc::channel(1024);
 
 		let pool = transaction_pool.pool().clone();
-		let import_stream = pool.validated_pool().import_notification_stream().map(|_| {
-			sc_consensus_manual_seal::rpc::EngineCommand::SealNewBlock {
+		let import_stream = pool
+			.validated_pool()
+			.import_notification_stream()
+			.filter(move |_| futures::future::ready(is_instant))
```

Comment:
This feels like an odd way to disable the stream, but just filtering everything. A more explicit if statement feels a bit more clear as to what it is doing.

```
let commands_stream = if is_instant {
  let pool = transaction_pool.pool().clone();
  let import_stream = pool.validated_pool()
  // ...
  futures::stream_select!(commands_stream, import_stream)
} else {
  commands_stream
}

Reply:
I originally tried that, but the rust compiler would not allow it since it would mean the type of `commands_stream` would be different depending on the value of `is_instant`. 

The `stream_select!` macro returns a different type than `mpsc::channel` -- While both conform to the `Stream<_>` interface, Rust does not seem like it plays well with traits being passed around.

Reply:
Ah and `commands_stream` type is not an easy one to just manually set. I guess that works. Feels like there is an easier way we are missing, but this is simple.